src/openssl_wrap.cpp

```diff
@@ -440,6 +440,10 @@ X509_VERIFY_PARAM* _X509_STORE_CTX_get0_param(X509_STORE_CTX *ctx) {
     return OpensslCallPtr::callChecked(lib::OpenSSLLib::SSL_X509_STORE_CTX_get0_param, ctx);
 }
 
+void _X509_check_ca(X509 *cert) {
+    OpensslCallIsPositive::callChecked(lib::OpenSSLLib::SSL_X509_check_ca, cert);
```

Comment:
The callChecked facility is meant for error handling. If I have a look at the openssl code, it looks like this function does not return error codes but 0 if it is not a CA and other positive codes depending on which kind of CA was detected. Hence, I would propose to make this function return a boolean (other things are not of interest to us anyways) and return true iff the call returns a positive number.

Reply:
Understood. Corrected on new version.